bazel/python_rules.bzl

```diff
@@ -22,41 +22,75 @@ load(
     "get_plugin_args",
     "get_proto_arguments",
     "includes_from_deps",
+    "is_in_virtual_imports",
+    "is_well_known",
     "protos_from_context",
 )
 
 _GENERATED_PROTO_FORMAT = "{}_pb2.py"
 _GENERATED_GRPC_PROTO_FORMAT = "{}_pb2_grpc.py"
 
-def _generate_py_impl(context):
-    protos = protos_from_context(context)
-    includes = includes_from_deps(context.attr.deps)
+PyProtoInfo = provider(
+    "The Python outputs from the Protobuf compiler.",
+    fields = {
+        "py_info": "A PyInfo provider for the generated code.",
+        "generated_py_srcs": "The direct (not transitive) generated Python source files.",
+    },
+)
+
+def _get_staged_proto_file(target, context, source_file):
```

Comment:
Is there any difference between this and the function in `protobuf.bzl`? I don't think I can see any. If not, I'd recommend removing the underscore prefix and just using the definition in `protobuf.bzl`.